src/game/common/modules/updatemodule.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+ * @file
+ *
+ * @author xezon
+ *
+ * @brief
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "updatemodule.h"
+
+#include "gamelogic.h"
+#include "thing.h"
+
+UpdateModule::UpdateModule(Thing *thing, const ModuleData *module_data) :
+    BehaviorModule(thing, module_data),
+    m_updatePhase(0), // ???
+    m_index(-1) // ???
+{
+}
+
+void UpdateModule::CRC_Snapshot(Xfer *xfer)
+{
+    BehaviorModule::CRC_Snapshot(xfer);
+}
+
+void UpdateModule::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t version = 1;
+    xfer->xferVersion(&version, 1u);
+    BehaviorModule::Xfer_Snapshot(xfer);
+
+    xfer->xferUnsignedInt(&m_updatePhase);
+
+    if (xfer->Get_Mode() == XFER_LOAD) {
+        m_updatePhase &= 0xFFFFFFFC;
+        m_updatePhase |= Get_Update_Phase();
+
+        if (m_updatePhase == 0) {
+            Encode_Frame(g_theGameLogic->Get_Frame());
+        }
+        m_index = -1;
+    }
+}
+
+void UpdateModule::Load_Post_Process()
+{
+    BehaviorModule::Load_Post_Process();
+}
+
+UpdateModuleInterface *UpdateModule::Get_Update()
+{
+    return static_cast<UpdateModuleInterface *>(this);
+}
+
+SleepyUpdatePhase UpdateModule::Get_Update_Phase() const
+{
+    return SLEEPY_UPDATE_PHASE_2;
+}
+
+BitFlags<DISABLED_TYPE_COUNT> UpdateModule::Get_Disabled_Types_To_Process() const
+{
+    return BitFlags<DISABLED_TYPE_COUNT>();
+}
+
+void UpdateModule::Set_Index(int index)
+{
+    m_index = index;
+}
+
+int UpdateModule::Get_Index()
+{
+    return m_index;
+}
+
+void UpdateModule::Encode_Frame(unsigned int frame)
+{
+    if (frame > 0x3FFFFFFFu) {
+        frame = 0x3FFFFFFFu;
+    }
+    m_updatePhase = Get_Update_Phase();
+    m_updatePhase |= (frame << 2);
+}
+
+unsigned int UpdateModule::Decode_Frame() const
+{
+    return m_updatePhase >> 2;
+}
+
+unsigned int UpdateModule::Get_Wake_Frame() const
```

Comment:
this technically returns UpdateSleepTime, i guess to prevent misuse they used a enum

Reply:
Fixed.